### PR TITLE
Use updated version of ex_admin, upgrade to Phoenix 1.4.3

### PR DESCRIPTION
**Pull request by tspenov**

### CODE REVIEW

File: config/dev.exs [hunk 1]
```diff
@@ -38,6 +47,7 @@ config :logger, :console, format: "[$time][$level][$metadata] $message\n"
 # Set a higher stacktrace during development. Avoid configuring such
 # in production as building large stacktraces may be expensive.
 config :phoenix, :stacktrace_depth, 20
+# json_library: Jason
```
Comment (IvanIvanoff) [inline config/dev.exs]: remove comment

File: lib/sanbase_web/channels/user_socket.ex [hunk 2]
```diff
@@ -5,7 +5,7 @@ defmodule SanbaseWeb.UserSocket do
   # channel "room:*", SanbaseWeb.RoomChannel
 
   ## Transports
-  transport(:websocket, Phoenix.Transports.WebSocket)
+  # transport(:websocket, Phoenix.Transports.WebSocket)
```
Comment (IvanIvanoff) [inline lib/sanbase_web/channels/user_socket.ex]: Remove if not needed

File: lib/sanbase_web/router.ex [hunk 3]
```diff
@@ -88,10 +88,6 @@ defmodule SanbaseWeb.Router do
       plug(:accepts, ["html"])
       plug(:put_secure_browser_headers)
     end
-
-    scope "/" do
-      get("/*path", ReverseProxy, upstream: ["http://localhost:3000"])
```
Comment (IvanIvanoff) [inline lib/sanbase_web/router.ex]: we would probably need that if we want to start frontend locally to a local backend :?

File: mix.exs [hunk 4]
```diff
@@ -37,20 +37,19 @@ defmodule Sanbase.Mixfile do
...
-      {:phoenix, "~> 1.3.0"},
-      {:phoenix_pubsub, "~> 1.0"},
-      {:phoenix_ecto, "~> 3.2"},
+      {:phoenix, "~> 1.4.3"},
+      {:phoenix_pubsub, "~> 1.1"},
+      {:phoenix_ecto, "~> 3.3"},
       {:postgrex, ">= 0.0.0"},
       {:gettext, "~> 0.11"},
-      {:cowboy, "~> 1.0"},
+      {:cowboy, "~> 2.0"},
```
Comment (IvanIvanoff) [inline mix.exs]: :tada:

File: mix.exs [hunk 5]
```diff
@@ -37,20 +37,19 @@ defmodule Sanbase.Mixfile do
 
   defp deps() do
     [
-      {:phoenix, "~> 1.3.0"},
-      {:phoenix_pubsub, "~> 1.0"},
-      {:phoenix_ecto, "~> 3.2"},
+      {:phoenix, "~> 1.4.3"},
```
Comment (IvanIvanoff) [inline mix.exs]: :rocket:

File: mix.exs [hunk 6]
```diff
@@ -89,7 +88,7 @@ defmodule Sanbase.Mixfile do
       {:jason, "~> 1.1"},
       {:elasticsearch, "~> 0.5"},
       {:quantum, "~> 2.3"},
-      {:plug_cowboy, "~> 1.0"},
+      {:plug_cowboy, "~> 2.0"},
```
Comment (IvanIvanoff) [inline mix.exs]: :robot:

File: assets/js/app.js [hunk 7]
```diff
@@ -19,3 +19,4 @@ import "phoenix_html"
 // paths "./socket" or full ones "web/static/js/socket".
 
 // import socket from "./socket"
+import css from '../css/app.css';
```
[the remark this answers is not in the document]
Comment (IvanIvanoff) [reply assets/js/app.js]: This change is according to the official phoenix 1.4 migration guide: https://gist.github.com/chrismccord/bb1f8b136f5a9e4abc0bfc07b832257e#css-changes

[the remark this answers is not in the document]
Comment (kalkov) [reply assets/js/app.js]: Weird stuff :see_no_evil:

File: lib/sanbase_web/router.ex (hunk 3 above)
Comment (tspenov) [reply lib/sanbase_web/router.ex]: Reverse proxy needs cowboy 1.0 :( https://github.com/slogsdon/elixir-reverse-proxy/blob/master/mix.exs#L26 Have to find another way around this.

File: lib/sanbase_web/router.ex [hunk 8]
```diff
@@ -78,21 +78,7 @@ defmodule SanbaseWeb.Router do
     pipe_through(:browser)
 
     get("/consent", RootController, :consent)
-    get("/apiexamples", ApiExamplesController, :api_examples)
```
Comment (IvanIvanoff) [inline lib/sanbase_web/router.ex]: why do you remove the api examples?

Comment (tspenov) [reply lib/sanbase_web/router.ex]: It is not used anymore. It was for `docs.santiment.net` but then it was changed with https://github.com/santiment/san-api, now this is used: https://help.santiment.net/metrics-explained#api.
Also you can check in sanbase logs for `"GET /apiexamples"`

File: mix.exs (hunk 4 above)
Comment (IvanIvanoff) [reply mix.exs]: you can even write ~> 2.6

**Pull request merged**